### **Entendimento do Problema:**

Melhorar a conversão de vendas de uma determinada empresa a partir da análise de campanhas realizadas no Facebook.

Fonte dos dados: https://www.kaggle.com/loveall/clicks-conversion-tracking

#### Importando bibliotecas de interesse

In [ ]:
# Importação e Manipulação do .CSV
import pandas as pd

#### Análise, Interpretação e Visualização de Dados

Carregando os dados:

In [ ]:
# Criando uma variável que recebe o dataframe
df = pd.read_csv("KAG_conversion_data.csv")
type(df)

pandas.core.frame.DataFrame

Primeira Inspeção:

In [ ]:
df.head() # Por default, exibe as 5 primeiras linhas do dataframe.

ad_id  xyz_campaign_id  ...  Total_Conversion Approved_Conversion
0  708746              916  ...                 2                   1
1  708749              916  ...                 2                   0
2  708771              916  ...                 1                   0
3  708815              916  ...                 1                   0
4  708818              916  ...                 1                   1

[5 rows x 11 columns]

In [ ]:
df.tail() # Por default, exibe as 5 últimas linhas do dataframe.

ad_id  xyz_campaign_id  ...  Total_Conversion Approved_Conversion
1138  1314410             1178  ...                13                   2
1139  1314411             1178  ...                 3                   0
1140  1314412             1178  ...                 2                   0
1141  1314414             1178  ...                 8                   2
1142  1314415             1178  ...                 5                   2

[5 rows x 11 columns]

In [ ]:
df.columns # Lista todas as colunas do dataframe.

Index(['ad_id', 'xyz_campaign_id', 'fb_campaign_id', 'age', 'gender',
       'interest', 'Impressions', 'Clicks', 'Spent', 'Total_Conversion',
       'Approved_Conversion'],
      dtype='object')

In [ ]:
df.index

RangeIndex(start=0, stop=1143, step=1)

In [ ]:
df.shape

(1143, 11)

In [ ]:
linhas, colunas = df.shape
print(f"Número de linhas: {linhas}")
print(f"Número de colunas: {colunas}")

Número de linhas: 1143
Número de colunas: 11


In [ ]:
serie_interest = df["interest"]
type(serie_interest)

pandas.core.series.Series

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1143 entries, 0 to 1142
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ad_id                1143 non-null   int64  
 1   xyz_campaign_id      1143 non-null   int64  
 2   fb_campaign_id       1143 non-null   int64  
 3   age                  1143 non-null   object 
 4   gender               1143 non-null   object 
 5   interest             1143 non-null   int64  
 6   Impressions          1143 non-null   int64  
 7   Clicks               1143 non-null   int64  
 8   Spent                1143 non-null   float64
 9   Total_Conversion     1143 non-null   int64  
 10  Approved_Conversion  1143 non-null   int64  
dtypes: float64(1), int64(8), object(2)
memory usage: 98.4+ KB


*Nota: Podemos perceber que esse dataframe é um conjunto sem dados faltantes; ou seja, todas as colunas possuem valores para cada uma das 1143 linhas do dataframe.*

**Dicionário dos Dados:**

- *ad_id*: ID única de cada anúncio.
- *xyz_campaign_id*: ID associada a cada anúncio da empresa XYZ.
- *fb_campaign_id*: ID associada a como o Facebook identifica cada campanha.
- *age*: idade da pessoa a qual o anúncio é entregue.
- *gender*: gênero da pessoa a qual o anúncio é entregue.
- *interest*: um código que especifica a categoria de interesse da pessoa (como mencionado no perfil público da pessoa no Facebook).
- *Impressions*: o número de vezes que o anúncio foi exibido.
- *Clicks*: número de cliques no anúncio.
- *Spent*: quantia paga ao Facebook (pela empresa XYZ) pela exibição do anúncio.
- *Total_Conversion*: Número total de pessoas que perguntaram sobre o produto após visualizar do anúncio.
- *Approved_Conversion*: Número total de pessoas que compraram o produto após visualizar o anúncio.
